snuba/clickhouse/columns.py

```diff
@@ -76,7 +79,7 @@ def __init__(self, inner_type: ColumnType) -> None:
     def get_all_modifiers(self) -> Iterable[Type[ColumnTypeWithModifier]]:
         def get_nested_modifiers(
             obj: ColumnType,
-        ) -> Optional[Iterable[Type[ColumnTypeWithModifier]]]:
+        ) -> Iterable[Type[ColumnTypeWithModifier]]:
             if not isinstance(obj, ColumnTypeWithModifier):
                 return []
```

Comment:
This algorithm is not supposed to work this way anymore since every type class has a valid get_all_modifiers.
```
if not isinstance(obj, ColumnTypeWithModifier):
   return obj.get_all_modifiers()
else:
  nested_modifiers = get_nested_modifiers(obj.inner_type)
  nested_modifiers.append(type(obj))
  return nested_modifiers